### PR TITLE
Home-Login-Dashboard

### PR DESCRIPTION
**Pull request by JANHMS**
### Description:

This PR creates Login, Authentication functionality with BE AUTH check and redirects to the users home. This gives the general flow when a user logs in into the app and gets the first touch with the app itself.

### Deploy:

This might be committed at last due to the fact it may have lots of merge conflicts with other open PR's.

### CODE REVIEW

**Comment by noramurakozy** [comment]
Is it normal that for me it looks like this?
I could register btw, so no problem there, only with the look
image
image

**Comment by JANHMS** [comment]
@noramurakozy no it is not normal

File: src/pages/UploadCSV.js [hunk 1]
```diff
@@ -83,6 +87,11 @@ class UploadCSV extends Component {
     };
     saveUserSleep(file).then( () => toast("file saved"))
     reader.readAsBinaryString(file);
+    
+    firestore.collection("users")
+    .doc(userId)
+    .set({dataUploaded: true})
```
Comment (JANHMS) [inline src/pages/UploadCSV.js:93]: here we set userData Uploaded to true, so we could track that for the redirect

**Comment by JANHMS** [comment]
this PR has been pulled and reworked here 

https://github.com/JANHMS/SleepDietApp/pull/15

**Pull request closed without merging**